### PR TITLE
Batch setup

### PR DESCRIPTION
**Pull request by yellowcap**
This is the version that we used to run our first deployment at scale.

### What changes
Main pipeline logic changes:

- Only concatenate S2, DEM, and S1 arrays for each tile during tililng. This saves lots of memory as S2 can be kept in Uint16.
- Use indexing on the arrays instead of the `.sel` function. Indexing is almost instant, and `sel` is slow for large arrays.
- Make the target bucket name an argument

Organisational changes:

- Add info about how to run the pipeline on batch.
- Add mgrs sample data as geopackage.

### CODE REVIEW

File: scripts/tile.py [hunk 1]
```diff
@@ -84,22 +78,18 @@ def tiler(stack, date, mgrs):
...
-                    y=slice(
-                        stack.y.values[y_start],
-                        stack.y.values[y_end]
-                        + np.sign(stack.rio.transform()[0]) * EPSILON,
-                    ),
-                )
+                parts = [part[:, y_start:y_end, x_start:x_end] for part in stack]
+
+                # Only concat here to save memory, it converts S2 data to float
+                tile = xr.concat(parts, dim="band").rename("tile")
```
Comment (weiji14) [inline scripts/tile.py:84]: @yellowcap or @lillythomas, can you double check the Sentinel-1 pixel values? I've been getting NaN loss values, and tracked it down to the Sentinel-1 VV and VH bands being all NaNs. Sentinel-2 and DEM bands are ok. Sample code:

```python
import os
import matplotlib.pyplot as plt
import rioxarray
import xarray as xr

os.environ["AWS_PROFILE"] = "xxx"

da = rioxarray.open_rasterio(
    filename="s3://clay-tiles-01/01/56HKH/2022-06-18/claytile-56HKH-2022-06-18-01-1.tif"
)
da = da.compute()

# %%
# Plot different bands
da.sel(band=1).plot.imshow()  # Blue
plt.show()
da.sel(band=10).plot.imshow()  # SWIR2
plt.show()
da.sel(band=11).plot.imshow()  # VH
plt.show()
da.sel(band=12).plot.imshow()  # VV
plt.show()
da.sel(band=13).plot.imshow()  # DEM
plt.show()
```

Blue band
SWIR2 band
VH band
VV band
DEM band

Comment (weiji14) [reply scripts/tile.py:84]: Tried a few other tiles in other locations like MGRS 29SMD over Lisbon, and it seems like the DEM can come up with NaNs as well (in places over the ocean?). So maybe check through all those bands.

Comment (yellowcap) [reply scripts/tile.py:84]: Good catch! Looks like the [nodata filter](https://github.com/Clay-foundation/model/blob/f48c01520099d7ef7ea916008573be3959c176a9/scripts/tile.py#L27) is missing checks on S1 and DEM. That should be easy to correct.

Comment (weiji14) [reply scripts/tile.py:84]: Ok, look like Lilly will be fixing this in #60.

File: scripts/tile.py [hunk 2]
```diff
@@ -84,22 +78,18 @@ def tiler(stack, date, mgrs):
...
 
                 if not filter_clouds_nodata(tile):
                     continue
 
+                counter += 1
+                if counter % 100 == 0:
+                    print(f"Counted {counter} tiles")
```
[the remark this answers is not in the document]
Comment (weiji14) [reply scripts/tile.py:91]: Alternatively, if we can store the actual timestamp in the metadata of the GeoTIFF, that would be even better!

File: scripts/mgrs_sample.gpkg
Comment (weiji14) [inline scripts/mgrs_sample.gpkg:1]: Should this geopackage be stored on git? Or just on the s3 bucket?

File: scripts/tile.py [hunk 3]
```diff
@@ -115,12 +105,8 @@ def tiler(stack, date, mgrs):
                 with rasterio.open(name, "r+") as rst:
```
Comment (weiji14) [inline scripts/tile.py:105]: At about L102 on the filename, @mattpaul had some suggestions at https://github.com/Clay-foundation/model/issues/35#issuecomment-1841585520 to:

1. Use underscore as the delimiter between MGRS/DATE/VERSION/COUNTER
2. Preface the VERSION with a v (e.g. `v0`)

Also mentioned at https://github.com/Clay-foundation/model/pull/44#discussion_r1399732116 that we could drop the hyphen in the date, i.e. YYYY-MM-DD becomes YYYYMMDD, but that didn't seem to have been applied in the initial batch.

File: scripts/tile.py (hunk 1 above)
Comment (yellowcap) [reply scripts/tile.py:84]: converted to issue for tracking https://github.com/Clay-foundation/model/issues/68

File: scripts/tile.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (yellowcap) [reply scripts/tile.py:91]: https://github.com/Clay-foundation/model/issues/69
https://github.com/Clay-foundation/model/issues/70

File: scripts/mgrs_sample.gpkg
Comment (yellowcap) [reply scripts/mgrs_sample.gpkg:1]: https://github.com/Clay-foundation/model/issues/71

File: scripts/tile.py (hunk 3 above)
Comment (yellowcap) [reply scripts/tile.py:105]: ok thanks will take that into consideration before next run.

**Comment by yellowcap** [comment]
Converted remaining questions to issues and will merge now for the sake of moving fast.

**Pull request merged**